### PR TITLE
Enable third party defi positions & fix token list exclusion logic

### PR DESCRIPTION
**Pull request by maxbbb**
Fixes APP-####

## What changed (plus any additional context for devs)
- re-enables third party positions
- removes staked deposits from token list exclusion
- only adds deposit to exclusion list if asset.defi_position is true
- fix NaN% for LP pools where total assets value is 0. 

## Screenshots
Before
Screenshot 2024-12-05 at 2 51 55 PM

After
IMG_0416

## How to Test
- if you have defi positions in your wallet, ensure if you have a token deposited in one of those positions and also have that token in your wallet that it is not missing in the main token list. If you don't a wallet with defi positions, grunt.eth is a good one.
- Check any pool positions such as Uniswap for incorrect asset allocation percentages. grunt.eth has a uniswap position that originally surfaced the bug related to both assets in the pool being worth 0.

### CODE REVIEW

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@kane/fix-thirdparty-positions/7a26346f4096a6be2f2214dda6f939f41182a54f.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@kane/fix-thirdparty-positions/7a26346f4096a6be2f2214dda6f939f41182a54f.ipa) for 7a26346f4096a6be2f2214dda6f939f41182a54f

**Comment by derHowie** [comment]
hey @maxbbb, could you throw a before and after on here, please? Some testing instructions and an address or two that you know are good test cases would also be helpful.

File: src/resources/defi/utils.ts
Comment (walmat) [inline src/resources/defi/utils.ts:1]: growing really worried about the performance of these calculations in this file but this is fine for now and can be touched up later

**Review by walmat — APPROVED** [review]

**Comment by maxbbb** [comment]
> hey @maxbbb, could you throw a before and after on here, please? Some testing instructions and an address or two that you know are good test cases would also be helpful.

added some screenshots and testing context

File: src/resources/defi/utils.ts
Comment (maxbbb) [reply src/resources/defi/utils.ts:1]: definitely some easy optimizations to be had can cleanup in a later pr

**Review by derHowie — APPROVED** [review]

**Pull request merged**